src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2851,6 +2882,123 @@ NetworkOPsImp::pubProposedTransaction(
     pubAccountTransaction(lpCurrent, alt, false);
 }
 
+void
+NetworkOPsImp::forwardProposedTransaction(Json::Value const& jvObj)
+{
+    // reporting does not forward validated transactions
+    // validated transactions will be published to the proper streams when the
+    // etl process writes a validated ledger
+    if (jvObj[jss::validated].asBool())
+        return;
+    {
+        std::lock_guard sl(mSubLock);
+
+        auto it = mStreamMaps[sRTTransactions].begin();
+        while (it != mStreamMaps[sRTTransactions].end())
+        {
+            InfoSub::pointer p = it->second.lock();
+
+            if (p)
+            {
+                p->send(jvObj, true);
+                ++it;
+            }
+            else
+            {
+                it = mStreamMaps[sRTTransactions].erase(it);
+            }
+        }
+    }
+
+    forwardProposedAccountTransaction(jvObj);
+}
+
+void
+getAccounts(Json::Value const& jvObj, std::vector<AccountID>& accounts)
```

Comment:
I also prefer return values to out params, but this is an out param because the function is called recursively. I'm not sure how to avoid unnecessary copies without using an out param.